mesonbuild/modules/pkgconfig.py

```diff
@@ -526,7 +526,7 @@ def parse_variable_list(stringlist):
             for lib in deps.pub_libs:
                 if not isinstance(lib, str) and not hasattr(lib, 'generated_pc'):
                     lib.generated_pc = filebase
-                    location = types.SimpleNamespace(subdir=state.subdir,
+                    location = types.SimpleNamespace(filename=state.subdir,
                                                      lineno=state.current_lineno)
```

Comment:
Seems weird, I guess filename should be a file and not a dir. If I understand correctly it should be `filename = os.path.join(state.subdir, environment.build_filename)`, no?

Reply:
@jon-turney ^

Reply:
Yes.  It seems like I missed an instance of this, despite checking several time.

Reply:
Perhaps the ModuleState passed to a module snippet should have a location member we would use here instead, as a more general solution.

Not being caught earlier also indicates that this warning doesn't have any test coverage.

Reply:
I added the `build_filename` to the `ModuleState` but I was wondering if the build_filename can be different from `meson.build`